IHP/IDE/Data/Controller.hs

```diff
@@ -41,9 +41,15 @@ instance Controller DataController where
 
     action ShowQueryAction = do
         connection <- connectToAppDb
-        let query = (param @Text "query")
-        when (query == "") $ redirectTo ShowDatabaseAction
-        rows :: [[DynamicField]] <- if isQuery query then PG.query_ connection (fromString (cs query)) else PG.execute_ connection (fromString (cs query)) >> return []
+        let queryText = param @Text "query"
+        when (queryText == "") $ redirectTo ShowDatabaseAction
+        let query = fromString $ cs queryText
+
+        queryResult :: Either PG.SqlError [[DynamicField]] <- if isQuery queryText then
+                (PG.query_ connection query <&> Right) `catch` (pure . Left)
+            else
+                PG.execute_ connection query >> pure (Right []) `catch` (pure . Left)
```

Comment:
I have no idea if this is the best way to catch an sql error, I thought it would be nice if it was converted to an `Either` so it's easier to show in the UI, feel free to advise something better if this is bad :slightly_smiling_face:

Reply:
Looks good to me 👍